### ISSUE TITLE
GNOME/GTK:  Librsvg BuildStream GTK GStreamer

### ISSUE THREAD

**Issue by sudocurse** [sudocurse/reading-list#5044]
GNOME/GTK: Librsvg, BuildStream, GTK, GStreamer<br>
http://ift.tt/2FGJcHI<br>
<ul>
<li>
<h5><a href="https://people.gnome.org/~federico/blog/rsvg-bench.html" rel="nofollow">rsvg-bench - a benchmark for librsvg</a></h5>
<blockquote>
<p>Librsvg 2.42.0 came out with a rather major performance regression compared to 2.40.20: SVGs with many transform attributes would slow it down. It was fixed in 2.42.1. We changed from using a parser that would recompile regexes each time it was called, to one that does simple string-based matching and parsing.</p>
<p>When I rewrote librsvg's parser for the transform attribute from C to Rust, I was just learning about writing parsers in Rust. I chose lalrpop, an excellent, Yacc-like parser generator for Rust. It generates big, fast parsers, like what you would need for a compiler &mdash; but it compiles the tokenizer's regexes each time you call the parser. This is not a problem for a compiler, where you basically call the parser only once, but in librsvg, we may call it thousands of times for an SVG file with thousands of objects with transform attributes.</p>
<p>So, for 2.42.1 I rewrote that parser using rust-cssparser. This is what Servo uses to parse CSS data; it's a simple tokenizer with an API that knows about CSS's particular constructs. This is exactly the kind of data that librsvg cares about. Today all of librsvg's internal parsers work using rust-cssparser, or they are so simple that they can be done with Rust's normal functions to split strings and such.</p>
</blockquote>
</li>
<li>
<h5><a href="https://blogs.gnome.org/tvb/2018/02/03/buildstream-hackfest-and-fosdem/" rel="nofollow">BuildStream Hackfest and FOSDEM</a></h5>
<blockquote>
<p>I also wanted to sum up a last minute BuildStream hackfest which occurred in Manchester just a week ago. Bloomberg sent some of their Developer Experience engineering team members over to the Codethink office in Manchester where the whole BuildStream team was present, and we split up into groups to plan upcoming coding sprints, land some outstanding work and fix some bugs.</p>
</blockquote>
</li>
<li>
<h5><a href="https://blogs.gnome.org/otte/2018/02/03/builders/" rel="nofollow">builders</a></h5>
<blockquote>
<p>An idiom that has shown up in GTK4 development is the idea of immutable objects and builders. The idea behind an immutable object is that you can be sure that it doesn&rsquo;t change under you, so you don&rsquo;t need to track changes, you can expose it in your API without having to fear users of the API are gonna change that object under you, you can use it as a key when caching and last but not least you can pass it into multiple threads without requiring synchronization.<br>
Examples of immutable objects in GTK4 are GdkCursor, GdkTexture, GdkContentFormats or GskRenderNode.</p>
</blockquote>
</li>
<li>
<h5><a href="https://blogs.gnome.org/mclasen/2018/02/03/gtk-hackfests-day-2/" rel="nofollow">GTK+ hackfest, day 2</a></h5>
<blockquote>
<p>The second day of the GTK+ hackfest in Brussels started with an hour of patch review. We then went through scattered items from the agenda and collected answers to some questions.</p>
</blockquote>
</li>
<li>
<h5><a href="https://www.phoronix.com/scan.php?page=news_item&amp;px=GTK4-2018-Fall-Target" rel="nofollow">GTK+ 4.0 Targeted For Its Initial Release This Fall, GTK+ 5.0 Development To Follow</a></h5>
<blockquote>
<p>A few days back I wrote about how GTK+ 4.0 is being talked about for release this year and now a bit more specific timeline is in place.</p>
<p>The past few days prior to FOSDEM in Brussels was a GTK+ hackfest. Among the items discussed when not banging on code was a GTK+ 4.0 road-map and coming out of this event in Belgium is a more solid understanding now that the initial GTK+ 4.0 release will be targeted for the fall of this year. There isn't any firm release plan at this time but at GUADEC (taking place in Spain this summer) they will revisit their plans to verify they can still ship this fall.</p>
</blockquote>
</li>
<li>
<h5><a href="http://blog.nirbheek.in/2018/02/gstreamer-webrtc.html" rel="nofollow">GStreamer has grown a WebRTC implementation</a></h5>
<blockquote>
<p>Late last year, we at Centricular announced a new implementation of WebRTC in GStreamer. Today we're happy to announce that after community review, that work has been merged into GStreamer itself! The plugin is called webrtcbin, and the library is, naturally, called gstwebrtc.</p>
<p>The implementation has all the basic features, is transparently compatible with other WebRTC stacks (particularly in browsers), and has been well-tested with both Firefox and Chrome.</p>
</blockquote>
</li>
<li>
<h5><a href="https://www.phoronix.com/scan.php?page=news_item&amp;px=GStreamer-WebRTC-Lands" rel="nofollow">GStreamer Lands A WebRTC Plugin</a></h5>
<blockquote>
<p>The GStreamer multimedia framework now has mainline support for WebRTC.</p>
<p>WebRTC is the set of protocols/APIs for real-time audio/video communication over peer-to-peer connections. WebRTC is supported by all major web browsers and more while now there is support within GStreamer too.</p>
</blockquote>
</li>
</ul>
<p><a href="http://www.tuxmachines.org/node/108747" target="_blank">read more</a></p>
<br>
<br>
via Tux Machines http://ift.tt/VTl7oF<br>
February 03, 2018 at 01:33PM
